packages/server-core/src/social/invite/invite.hooks.ts

```diff
@@ -45,14 +56,92 @@ import {
   inviteResolver
 } from './invite.resolvers'
 
+async function addSearch(context: HookContext<InviteService>) {
+  if (!context.params.query || !context.params.query.search) {
+    return
+  }
+
+  const search = context.params.query.search
+
+  context.params.query = {
+    ...context.params.query,
+    $or: [
+      {
+        inviteType: {
+          $like: '%' + search + '%'
+        }
+      },
+      {
+        passcode: {
+          $like: '%' + search + '%'
+        }
+      }
+    ]
+  }
+}
+
+function checkQueryType(type: InviteQuery['type']) {
+  return (context: HookContext<InviteService>) => context.params.query?.type === type
+}
+
+async function addInvitesReceived(context: HookContext<InviteService>) {
+  const identityProviders = (await context.app.service(identityProviderPath).find({
+    query: {
+      userId: context.params.query!.userId
+    }
+  })) as Paginated<IdentityProviderType>
+  const identityProviderTokens = identityProviders.data.map((provider) => provider.token)
+
+  context.params.query = {
+    ...context.params.query,
+    $or: [
+      {
+        inviteeId: context.params.query!.userId
+      },
+      {
+        token: {
+          $in: identityProviderTokens
+        }
+      }
+    ]
+  }
+}
+
+async function addInvitesSent(context: HookContext<InviteService>) {
```

Comment:
This hook seems redundant to me? Its just setting whats already there

Reply:
implemented!